### PR TITLE
cxx-qt-gen: add cxx_qt::qobject to determine which struct is the QObject

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Requires #195 #196

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,7 @@ mod my_object {
         }
     }
 
+    #[cxx_qt::qobject]
```
Comment (ahayzen-kdab) [inline README.md:88]: Need to document this in the book.

Comment (ahayzen-kdab) [reply README.md:88]: Done

**Review by LeonMatthesKDAB — CHANGES_REQUESTED** [review]
I'd want to see at least one full test from input->output that uses another name than RustObj.

Otherwise looks good :+1:

File: cxx-qt-gen/test_inputs/invokables.rs [hunk 2]
```diff
@@ -8,6 +8,7 @@ mod my_object {
         type QString = cxx_qt_lib::QString;
     }
 
+    #[cxx_qt::qobject]
     #[derive(Default)]
     pub struct RustObj;
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply cxx-qt-gen/test_inputs/invokables.rs:13]: The next pull request changes all the names away from RustObj, as it changes the QObject name to use the struct name and not the module name. So can we wait for that?

[the remark this answers is not in the document]
Comment (LeonMatthesKDAB) [reply cxx-qt-gen/test_inputs/invokables.rs:13]: Ah, alright, then just update the other comment and we can merge :+1:

File: cxx-qt-gen/src/extract.rs [hunk 3]
```diff
@@ -674,7 +674,7 @@ pub fn extract_qobject(module: &ItemMod) -> Result<QObject, TokenStream> {
     // The original Data Item::Struct if one is found
     let original_data_struct = qobject.data_struct;
     // The original RustObj Item::Struct if one is found
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply cxx-qt-gen/src/extract.rs]: Done

**Review by LeonMatthesKDAB — APPROVED** [review]
LGTM

**Pull request merged**